--- a/src/external_plugins/client-config.py
+++ b/src/external_plugins/client-config.py
@@ -18,2 +18,3 @@
-def normalize_client_config(config, plugins, frontend_domain, api_host):
+def normalize_client_config(config, plugins, abbr, frontend_domain, api_host):
+    config["abbr"] = abbr
     config["plugins"] = plugins
@@ -31 +32 @@
-        return normalize_client_config(json.loads(config), app.config.plugins, urlparse(app.config.frontend_urls.member).netloc, f"https://{app.config.domain}")
+        return normalize_client_config(json.loads(config), app.config.plugins, app.config.abbr, urlparse(app.config.frontend_urls.member).netloc, f"https://{app.config.domain}")
@@ -44 +45 @@
-    config = normalize_client_config(json.loads(t[0][0]), app.config.plugins, urlparse(app.config.frontend_urls.member).netloc, f"https://{app.config.domain}")
+    config = normalize_client_config(json.loads(t[0][0]), app.config.plugins, app.config.abbr, urlparse(app.config.frontend_urls.member).netloc, f"https://{app.config.domain}")
@@ -318,2 +319,3 @@
-        if frontend_conf.get("plugins") != config["plugins"] or frontend_conf.get("domain") != frontend_domain or frontend_conf.get("api_host") != api_host:
+        if frontend_conf.get("abbr") != config["abbr"] or frontend_conf.get("plugins") != config["plugins"] or frontend_conf.get("domain") != frontend_domain or frontend_conf.get("api_host") != api_host:
+            frontend_conf["abbr"] = config["abbr"]
             frontend_conf["plugins"] = config["plugins"]
