--- a/src/config.py
+++ b/src/config.py
@@ -588,6 +588,13 @@ def validateEmbed(embed):
     return embed
 
 def validateConfig(cfg):
+    if 'frontend_urls' not in cfg.keys() or type(cfg["frontend_urls"]) != dict:
+        cfg["frontend_urls"] = copy.deepcopy(default_config["frontend_urls"])
+    else:
+        for key in default_config["frontend_urls"].keys():
+            if key not in cfg["frontend_urls"].keys() or type(cfg["frontend_urls"][key]) != str:
+                cfg["frontend_urls"][key] = default_config["frontend_urls"][key]
+
     if 'hex_color' not in cfg.keys():
         cfg["hex_color"] = "2fc1f7"
     hex_color = cfg["hex_color"][-6:]
@@ -909,9 +916,6 @@ def validateConfig(cfg):
         cfg["member_accept"] = cfg["team_update"]
         del cfg["team_update"]
 
-    if 'email_confirm' not in cfg['frontend_urls'].keys():
-        cfg["frontend_urls"]["email_confirm"] = f"https://{cfg['domain']}/auth/email?secret={{secret}}"
-
     if 'server_host' not in cfg.keys() and "server_ip" in cfg.keys():
         cfg["server_host"] = cfg["server_ip"]
         del cfg["server_ip"]
@@ -1393,6 +1397,14 @@ def validateConfig(cfg):
         except:
             cfg["db_port"] = 3306
 
+    if 'domain' not in cfg.keys() or type(cfg["domain"]) != str:
+        cfg["domain"] = default_config["domain"]
+    if 'logo_url' not in cfg.keys() or type(cfg["logo_url"]) != str:
+        cfg["logo_url"] = default_config["logo_url"]
+    cfg["logo_url"] = cfg["logo_url"].replace("{domain}", cfg["domain"])
+    for key in default_config["frontend_urls"].keys():
+        cfg["frontend_urls"][key] = cfg["frontend_urls"][key].replace("{domain}", cfg["domain"])
+
     tcfg = {}
     for key in config_keys_order:
         if key in cfg.keys():
