--- a/makefile
+++ b/makefile
@@ -22,7 +22,7 @@
 	. $(VENV_DIR)/bin/activate && \
 	python3 -m nuitka src/main.py --output-dir=$(BUILD_DIR)/drivershub --output-filename=drivershub \
 		--standalone --include-package=websockets,tzdata --include-package-data=tzdata \
-		--show-progress --prefer-source-code
+		--show-progress --prefer-source-code --jobs=$(NUITKA_JOBS)
 	mkdir -p $(DIST_DIR)
 	cp -r $(BUILD_DIR)/drivershub/main.dist/* $(DIST_DIR)/
 
@@ -30,7 +30,7 @@
 	. $(VENV_DIR)/bin/activate && \
 	python3 -m nuitka src/bannergen/main.py --output-dir=$(BUILD_DIR)/bannergen \
 		--output-filename=bannergen --standalone --include-package=websockets \
-		--show-progress --prefer-source-code
+		--show-progress --prefer-source-code --jobs=$(NUITKA_JOBS)
 	mkdir -p $(DIST_DIR)
 	cp -r $(BUILD_DIR)/bannergen/main.dist/* $(DIST_DIR)/

